### ISSUE TITLE
Adding color by another dimension

### ISSUE THREAD

**Issue by kalehdoo** [michaeldmoore/michaeldmoore-scatter-panel#32]
My wishlist if not already considered.

Let's say I am plotting Height (X axis) and Weight (Y Axis). I want to change the color of the dots based on if the dot corresponds to a girl or a boy (Gender column).

**Comment by qootec** [comment]
Very similar request here.
My data has the following columns
xValue, yValue, Success

Creating the xValue vs yValue scatter plot is trivial.
Issue: I can't make sure the Success=True values are green and the Success=False are red.

I could perfectly divide the data in two separate queries:
- A-series: xValue,yValue (only those with True success)
- B-series: xValue,yValue (only those with False success)

It seems the x-Value is then only considered from the A-series. Those from the B-series are not considered.
Specifying the x-Value field from the second query doesn't show any values at all.
(To be noted: the x-Values found in series A and B are _not_ idential: the x-Values from the A-series do not occur in the B-series.)

Is this possible anyway or is this a current limitation?

**Comment by michaeldmoore** [comment]
It's been so long since I looked at this project, but I can think of a
simple way to do what you want.

If you arrange your data so that you have three columns something like this

X       Y_success           Y_fail
1       1                   (null)
2       (null)              6.3
3       3                   (null)
4       (null)              9.1
5       4.5                 (null)
6       (null)              2.1
7       6.3                 (null)
8       (null)              11.7
9       9.1                 (null)

etc.

That is, one row for all the X values, but only one of the two Y-ish
columns not-null,
then you can define the plot as two data series (one for the successes and
one for the failures, each with an appropriate color.  The null values
won't be plotted.

No need to sort the data like I'm showing here.  You can simply create a
union of the two data sets, or some other clever query
logic to create a data set like this that would work.

Let me know if you need a clearer explanation.

Michael Moore

**Comment by qootec** [comment]
Hi Michael,

Thanks for your suggestion. I didn't think about that possibility.
I obtain my data through a SPARQL query, so in my case (just as an example) I needed to formulate it similar to this:

```
SELECT  ?xValue  ?yValueOk ?yValueNotOk WHERE
{
    ?qc a :QCCheck;
        :feature ?xValue;
        :value ?qcValue.

    ?_ :CategoryOk ?categoryOk.

    optional {
        FILTER (?categoryOk)
        BIND (?qcValue as ?yValueOk)
    }
        
    optional {
        FILTER (! ?categoryOk)
       BIND (?qcValue as ?yValueNotOk)
    }
}
```

**Comment by francisozic** [comment]
I will add I would really appreciate this feature - the ability to put a heat map for the value of each point on a series based on a third value, e.g.
X Y M
-----
1 1 2
1 1 4

Typically this would be another color chart (legend) that corresponds to the "M" column value.  Plotly (deprecated) had this feature and was very, very useful for a quick look at three dimensional data.

**Comment by michaeldmoore** [comment]
It's been a while since I looked at this, but I think this capability might
already be in there.  I can't check it out right now (I'll take a look
tomorrow or in the next few days to make sure).

At least the ability to set the dot size corresponding to a numeric value
in another column of the data set.

I know I worked on adding a feature where the color could be defined in
another field - as a string, either a known color names such "red", "blue"
etc., or a rgb triad string like "rgb(23,54,23)".  Right now, I'm not 100%
sure this code made it into the release branch.

I'll get back to you in a day or three.  In the meantime, check out the
readme file or search the source code (in github) looking for the string
"rgb"
